### PR TITLE
fix: correct launch.json for node js debugging through vs code

### PR DESCRIPTION
**Pull request by TheSriram**
*Issue #, if available:*

https://github.com/awslabs/aws-sam-cli/issues/555

*Description of changes:*

stopOnEntry flag allows breaking directly into user code, instead of the runtime bootstrap file.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by heitorlessa** [comment]
AWESOME @TheSriram ;) 

Would you be able to update the PR to point out that this will work if the source code is in the root (where template.yaml is). However, if they have functions inside folders (i.e. function1/app.js, function2/app.js) then they'll need to specify that in `localroot` too. 

Here's an example to debug NodeJS when using the boilerplate provided by SAM init:

```bash
sam init
cd sam-app
# Brings Visual Studio code for this app
code . 
```

Creates a new debugger configuration using the following:

```json
```

File: docs/usage.rst [hunk 1]
```diff
@@ -308,21 +308,23 @@ the following launch configuration:
...
+		"configurations": [{
+	        "name": "Attach to SAM CLI",
+	        "type": "node",
+	        "request": "attach",
+	        "address": "localhost",
+	        "port": 5858,
+	        "localRoot": "${workspaceRoot}",
+	        "sourceMapPathOverrides": {
+	            "${workspaceRoot}/runtime": "${remoteRoot}"
+			    },
```
Comment (jfuss) [inline docs/usage.rst]: Can you format the JSON here?

File: docs/usage.rst [hunk 2]
```diff
@@ -308,21 +308,23 @@ the following launch configuration:
...
+	        "remoteRoot": "/var/task",
+	        "protocol": "inspector",
```
Comment (jfuss) [inline docs/usage.rst]: The previous protocol was legacy. What does updating this to inspector indicate?

File: docs/usage.rst [hunk 3]
```diff
@@ -308,21 +308,23 @@ the following launch configuration:
...
-   }
+	{
+		"version": "0.2.0",
+		"configurations": [{
+	        "name": "Attach to SAM CLI",
+	        "type": "node",
+	        "request": "attach",
+	        "address": "localhost",
+	        "port": 5858,
+	        "localRoot": "${workspaceRoot}",
```
Comment (jfuss) [inline docs/usage.rst]: This should be the ${workspaceRoot}/<path to the function code dir> right? This localRoot only works if you open the directory of the function code and not the whole project.

**Comment by TheSriram** [comment]
``
At what point does the previous config we had not work? Can we clearly document which version(s) of VSCode the config you added supports? We should probably keep the one one around too and say use that config if your VSCode version is lower than X?
``

It's a change in the way the first file the debugger tries to break into. we break into the code_uri directly with the stopOnEntry flag. Also this flag seems to exist from 2016, so it should definitely be backwards compatible. Yes this is VS code v1.27.2, I can add the version this was tested on.

**Pull request merged**